cyclic_boosting/base.py

```diff
@@ -887,8 +920,11 @@ def _check_stop_criteria(self, iterations: int, convergence_parameters: Converge
                 "analysis plots."
             )
 
+        if iterations <= 3 and self.hierarchical_feature_groups is not None:
```

Comment:
Rather ignorant question, why is 3 the magical number of iterations here? Is it a good balance between two criteria? 
If this is in fact a parameter one can tune, than 3 should be replaced by a constant, say `"TRAINING_ITERATIONS_HIER_FEATURES"` and the docstring above has to be adjusted.

Reply:
I think 3 is a good value, but a parameter with default 3 might be better, yes. I will change it.